catalog/from-k8s/resource.go

```diff
@@ -157,6 +161,13 @@ func (t *ServiceResource) shouldSync(svc *apiv1.Service) bool {
 		return false
 	}
 
+	// Ignore ClusterIP services if ClusterIP sync is disabled
+	if svc.Spec.Type == apiv1.ServiceTypeClusterIP {
```

Comment:
I think the filtering previously was a side effect of the service type switch statement in `generateRegistrations`. Putting it in the `shouldSync` method seems much cleaner, since we now support `ClusterIP` in `generateRegistrations`.